x/auth/ante/basic.go

```diff
@@ -80,8 +84,38 @@ func NewConsumeGasForTxSizeDecorator(ak keeper.AccountKeeper) ConsumeTxSizeGasDe
 }
 
 func (cgts ConsumeTxSizeGasDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	sigTx, ok := tx.(SigVerifiableTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "invalid tx type")
+	}
 	params := cgts.ak.GetParams(ctx)
 	ctx.GasMeter().ConsumeGas(params.TxSizeCostPerByte*sdk.Gas(len(ctx.TxBytes())), "txSize")
 
+	// simulate gas cost if the signature array is empty
+	if simulate && len(sigTx.GetSignatures()) == 0 {
+		for _, signer := range sigTx.GetSigners() {
+			acc := cgts.ak.GetAccount(ctx, signer)
+			pubkey := acc.GetPubKey()
+			// use placeholder simSecp256k1Pubkey if sig is nil
+			if pubkey == nil {
+				pubkey = simSecp256k1Pubkey
+			}
+			simSig := types.StdSignature{
```

Comment:
Unfortunately this decorator assumes that the signature is an `auth.StdSignature` which is not ideal.

Reply:
We already did this in the previous ant-handler. Let's clearly state this in the godoc.